lnd_services.go

```diff
@@ -766,3 +766,59 @@ func getClientConn(cfg *LndServicesConfig) (*grpc.ClientConn, error) {
 
 	return conn, nil
 }
+
+// GetTLSCredentials gets the tls credentials, whether provided as straight-up
+// data or a path to a certificate file.
+func GetTLSCredentials(tlsData, tlsPath string) (
+	credentials.TransportCredentials, error) {
+
+	if tlsPath != "" && tlsData != "" {
+		return nil, fmt.Errorf("must set only one: TLSPath or TLSData")
+	}
+
+	var creds credentials.TransportCredentials
+	var err error
+
+	// We'll determine if the tls certificate is passed in directly as
+	// data, by a path, or try the system's certificate chain, and then
+	// load it.
+	if tlsData != "" {
+		tlsBytes := []byte(tlsData)
+
+		block, _ := pem.Decode(tlsBytes)
+		if block == nil || block.Type != "CERTIFICATE" {
+			return nil, errors.New("failed to decode PEM block " +
+				"containing tls certificate")
+		}
+
+		cert, err := x509.ParseCertificate(block.Bytes)
+		if err != nil {
+			return nil, err
+		}
+
+		pool := x509.NewCertPool()
+		pool.AddCert(cert)
+
+		// Load the specified TLS certificate and build transport
+		// credentials.
+		creds = credentials.NewClientTLSFromCert(pool, "")
+	} else if tlsPath != "" {
+		if tlsPath == "" {
+			tlsPath = defaultTLSCertPath
+		}
+
+		creds, err = credentials.NewClientTLSFromFile(tlsPath, "")
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		pool, err := x509.SystemCertPool()
```

Comment:
Unfortunately, the `SystemCertPool()` call doesn't work on Windows. But there's a workaround for using the system cert pool in gRPC, see here: https://github.com/lightningnetwork/lnd/blob/master/cmd/lncli/main.go#L101